example/lib/main.dart

```diff
@@ -54,6 +60,20 @@ class _MyAppState extends State<MyApp> {
   @override
   Widget build(BuildContext context) {
     return MaterialApp(
+      routes: {
+        '/navDemo': (_) => const NavDemo(),
+      },
+      navigatorObservers: [
+        FSNavigatorObserver(
+          initialProperties: (current, previous) => {
+            if (current.settings.name == '/navDemo') 'navDemoFirstVisit': true,
+          },
+          updateProperties: (current, previous) => {
+            if (current.settings.name == '/navDemo') 'navDemoFirstVisit': false,
+            'navDemoLaterVisit': true,
+          },
+        )
+      ],
```

Comment:
Would it be better to put all of our pages here, and maybe skip the initial/update properties (and the manual `FS.page()`) calls in main.dart? 

It makes sense to me to have main.dart show the more normal usage, and then nav_demo.dart had the more esoteric things to ensure the full API is exercised.

Reply:
I think that's a good idea. I think it wants a separate PR, since we'll need to do a bit more refactoring, but worth doing.

We definitely want to show a few different ways of doing things, since the navigation story in Flutter is pretty fractured. I'll work on a follow up to clarify a couple different approaches with examples.